### PR TITLE
Add Back Navigation to Auth Screens

### LINKED ISSUE #1109: Add a way to go back during Login
### User Story

As a user, if I select Username / Password, or if I'm entering my phone number auth code, I should be able to go down the other route. 

### Description

If a user opts to enter their username and password, they have no way to go back to the option to type their phone number.

In the same way, if someone chooses to login via SMS, and they get to the screen where they need to enter their pin, they have no way to go back and change their phone number, or login via email / password. 

Designs are below :) 

back buttons

### Acceptance Criteria

* [ ] It should [SPEC_DETAIL]

### Other Resources

* Design mockup/prototype link
* Basecamp todo or link to "epic" on GitHub

### PR DESCRIPTION
**Pull request by erikkgagewhitt**
### What does this PR do, or why is it needed?

Adds back buttons to the Verification and Passwords screens pictured below. This will allow users to navigate back should they wish to choose another path or enter a different phone number.

### What design trade-offs/decisions were made?

@kevin-birdy Not so much a design decision as a callout; the Verification screen looks a little funky because of the stacking of the button on top of the logo on top of the title, so...

### How do I test this PR?

1. Launch the app
3. Choose to use your email to sign in
3. Tap the back button; you should return to the previous screen
4. Enter your phone number
3. Tap the back button; you should return to the previous screen

## SCREENSHOTS:

||IOS|Android|
|---|---|---|
|Verification|Screen Shot 2019-10-09 at 4 40 49 PM|Screen Shot 2019-10-09 at 4 46 51 PM|
|Password|Screen Shot 2019-10-09 at 4 41 26 PM|Screen Shot 2019-10-09 at 4 46 37 PM|

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes #1109 
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apollos-ui-auth/src/Password/index.js [hunk 1]
```diff
@@ -16,6 +22,23 @@ import LoginForm from './Login';
 
 import SignUpForm from './Signup';
 
+const styles = StyleSheet.create({
```
Comment (conrad-vanl) [inline packages/apollos-ui-auth/src/Password/index.js]: Man in some ways I miss the simplicity of `StyleSheet`

File: packages/apollos-ui-auth/src/Password/index.js [hunk 2]
```diff
@@ -16,6 +22,23 @@ import LoginForm from './Login';
...
+  backButton: {
+    marginRight: sizing.baseUnit,
+  },
+  backButtonContainer: {
+    width: sizing.baseUnit * 2,
+    height: sizing.baseUnit * 2,
```
Comment (conrad-vanl) [inline packages/apollos-ui-auth/src/Password/index.js]: Nice job finding `defaultTheme`! Hah. @burkeshartsis will show you `styled`, which is how we style components in Apollos.

**Comment by erikkgagewhitt** [comment]
@conrad-vanl  I looked everywhere for stylesheets and couldn't find any, so I knew I was doing something wrong. XD

**Comment by vinnyjth** [comment]
@burkeshartsis I'm going to leave you in charge of reviewing this :)

**Comment by vinnyjth** [comment]
@conrad-vanl or @burkeshartsis do you have time to look at this before EOD? If not, I'll take a look and get this merged.

**Comment by burkeshartsis** [comment]
I can look at it next. Maybe in 30min from now

File: packages/apollos-ui-auth/src/Password/index.js [hunk 3]
```diff
@@ -16,6 +19,22 @@ import LoginForm from './Login';
 
 import SignUpForm from './Signup';
 
+const StyledPaddedView = styled(
+  {
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  'ui-auth.Password.PaddedView'
```
Comment (vinnyjth) [inline packages/apollos-ui-auth/src/Password/index.js]: Nice work including the `name` for the styled component!

**Pull request merged**